conformance/lang/values-types-and-variables/nil-literal.balt

```diff
@@ -0,0 +1,173 @@
+Test-Case: output
+Description: Test nil type descriptor with nil literal '()'.
+Labels: nil-type-descriptor, nil-literal
+
+public function main() {
+    () x = ();
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil type descriptor with nil literal 'null'.
+Labels: nil-type-descriptor, nil-literal
+
+public function main() {
+    () x = null;
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test value and assignability of the nil literal.
+Labels: nil-type-descriptor, nil-literal, equality-expr
+
+public function main() {
+    () x = ();
+    () y = null;
+    io:println(x == y); // @output true
+
+    x = y;
+    io:println(x.toBalString()); // @output ()
+
+    x = ();
+    y = x;
+    io:println(y.toBalString()); // @output ()
```

Comment:
These aren't quite relevant to nil literal?

Reply:
Removed them.